plugins/out_pgsql/pgsql.c

```diff
@@ -0,0 +1,297 @@
+        pgsql_conf_destroy(ctx);
+        return -1;
+    }
+
+    flb_sds_destroy(ctx->db_table);
+    ctx->db_table = flb_sds_create(temp);
+    PQfreemem(temp);
+
+    if(!ctx->db_table) {
+        flb_errno();
+        pgsql_conf_destroy(ctx);
+        return -1;
+    }
+
+    flb_info("[out_pgsql] we check that the table %s "
+             "exists, if not we create it", ctx->db_table);
+
+    str_len = 62 + flb_sds_len(ctx->db_table);
+
+    query = flb_malloc(str_len);
+    if(query == NULL) {
+        flb_errno();
+        pgsql_conf_destroy(ctx);
+        return -1;
+    }
+
+    snprintf(query, str_len, "CREATE TABLE IF NOT EXISTS %s (data jsonb);",
+             ctx->db_table);
+    res = PQexec(ctx->conn, query);
+    flb_free(query);
+
+    if(PQresultStatus(res) != PGRES_COMMAND_OK) {
+        flb_error("[out_pgsql] %s", PQerrorMessage(ctx->conn));
+        pgsql_conf_destroy(ctx);
+        return -1;
+    }
+
+    PQclear(res);
+
+    flb_info("[out_pgsql] switching postgresql connection "
+             "to non-blocking mode");
+    if(PQsetnonblocking(ctx->conn, 1) != 0) {
+        flb_error("[out_pgsql] non-blocking mode not set");
+        pgsql_conf_destroy(ctx);
+        return -1;
+    }
+
+    return 0;
+}
+
+static void cb_pgsql_flush(const void *data, size_t bytes,
+                            const char *tag, int tag_len,
+                            struct flb_input_instance *i_ins,
+                            void *out_context,
+                            struct flb_config *config)
+{
+    struct flb_pgsql_config *ctx = out_context;
+    flb_sds_t json;
+    PGresult *res = NULL;
+    char *query = NULL;
+    size_t str_len;
+
+    if(PQconsumeInput(ctx->conn) == 0 && PQisBusy(ctx->conn) == 1) {
+        flb_debug("[out_pgsql] Some command may still be running");
+    }
+
+    /*
+      PQreset()
+      This function will close the connection to the server and attempt to
+      reestablish a new connection to the same server, using all the same
+      parameters previously used. This might be useful for error recovery
+      if a working connection is lost.
+     */
+    if(PQstatus(ctx->conn) != CONNECTION_OK) {
+        PQreset(ctx->conn);
+        FLB_OUTPUT_RETURN(FLB_RETRY);
+    }
+
+
+    json = flb_pack_msgpack_to_json_format(data, bytes,
+                                           FLB_PACK_JSON_FORMAT_JSON,
+                                           FLB_PACK_JSON_DATE_DOUBLE,
+                                           ctx->timestamp_key);
+    if(json == NULL) {
+        flb_errno();
+        flb_error("[]out_pgsql] Can't parse the msgpack into json");
+        FLB_OUTPUT_RETURN(FLB_RETRY);
+    }
+
+    str_len = 60 + flb_sds_len(json);
+    query = flb_malloc(str_len);
+
+    if (query == NULL) {
+        flb_errno();
+        flb_sds_destroy(json);
+        FLB_OUTPUT_RETURN(FLB_RETRY);
+    }
+
+    snprintf(query, str_len,
+             "INSERT INTO %s SELECT * FROM json_array_elements('%s');",
```

Comment:
I wonder if this is the line of code that causes the issue I saw?

Reply:
Yeah that's the line we changed to insert data... I'm using PostgreSQL 11 I'll try with that version later in the day

Reply:
with that version I mean "10.4" sorry not to be clear

Reply:
@PettitWesley @sxd Btw, I tried out 1.4.1, and ran into this same issue. I think the issue happens when there's a single quote in the `json`. I'm not super familiar with `C` or `fluent-bit`, but according to [this stack overflow issue](https://stackoverflow.com/a/46051720) I think the solution would be to replace all single quotes with two single quotes to escape them.

Reply:
Thank you Tucker! sadly the solution means that we may run into a few more issues so I expect to have an answer tomorrow =) will keep you post! 

Thanks again @tuckerconnelly

Reply:
Thanks Jonathan, looking forward to the fix--let me know if I can do any testing or help at all.